src/test/java/org/mifos/integrationtest/cucumber/stepdef/BulkPaymentStepDef.java

```diff
@@ -0,0 +1,168 @@
+                .when()
+                .post("/batchtransactions")
+                .andReturn().asString();
+        batchId = fetchBatchId(response);
+        logger.info(batchId);
+        logger.info("Batch transaction API response: " + response);
+    }
+
+    @Given("the batch ID for the submitted CSV file")
+    public void isBatchIdAvailable(){
+        assertThat(batchId).isNotEmpty();
+    }
+
+    @When("poll the batch summary API using the batch ID")
+    public void initiateBatchSummaryApi(){
+        int retries = 5;
+        int intervalInSeconds = 30;
+
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Platform-TenantId", "rhino");
+        RequestSpecification requestSpec = getDefaultSpec();
+
+        for(int i = 0; i < retries; i++) {
+            String response =  RestAssured.given(requestSpec)
+                    .baseUri("http://localhost:5060")
+                    .param("batchId", batchId)
+                    .param("X-CorrelationID", "12345678-6897-6798-6798-098765432134")
+                    .headers(headers)
+                    .expect()
+                    .spec(new ResponseSpecBuilder().expectStatusCode(200).build())
+                    .when()
+                    .get("/api/v1/batch")
+                    .andReturn().asString();
+            Gson gson = new Gson();
+            BatchSummaryResponse batchSummaryResponse = gson.fromJson(response, BatchSummaryResponse.class);
+
+            logger.info("Batch summary response: " + batchSummaryResponse.toString());
+
+            logger.info("Total is {}", batchSummaryResponse.getTotal());
+            logger.info("Ongoing is {}", batchSummaryResponse.getOngoing());
+            logger.info("Successful is {}", batchSummaryResponse.getSuccessful());
+            logger.info("Failed is {}", batchSummaryResponse.getFailed());
+
+            if(batchSummaryResponse != null && batchSummaryResponse.getTotal() != 0){
+                completionPercent = (int) (batchSummaryResponse.getSuccessful()/ batchSummaryResponse.getTotal() * 100);
+            }
+
+            try {
+                Thread.sleep(intervalInSeconds * 1000);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+    }
+
+    @Then("successful transactions percentage should be greater than or equal to minimum threshold")
+    public void batchSummarySuccessful(){
+        assertThat(completionPercent).isNotNull();
+        assertThat(completionPercent).isGreaterThan(thresholdPercent);
+    }
+
+    private static RequestSpecification getDefaultSpec() {
+        RequestSpecification requestSpec = new RequestSpecBuilder().build();
+        requestSpec.relaxedHTTPSValidation();
+        return requestSpec;
+    }
+
+    private MultiPartSpecification getMultiPart(String fileContent) {
+        return new MultiPartSpecBuilder(fileContent.getBytes()).
+                fileName("test.csv").
```

Comment:
Is this file coming in response? Otherwise where is the file provided

Reply:
@fynmanoj `fileContent` in this method already holds content of `bulk-payment.csv`, this file is already in  the PR. We are just setting the file name as `test.csv` here.